loopin.go

```diff
@@ -605,16 +633,29 @@ func (s *loopInSwap) waitForHtlcConf(globalCtx context.Context) (
 
 	notifier := s.lnd.ChainNotifier
 
-	confChanP2WSH, confErrP2WSH, err := notifier.RegisterConfirmationsNtfn(
-		ctx, s.htlcTxHash, s.htlcP2WSH.PkScript, 1, s.InitiationHeight,
-	)
+	notifyConfirmation := func(htlc *swap.Htlc) (
+		chan *chainntnfs.TxConfirmation, chan error, error) {
+
+		if htlc == nil {
```

Comment:
Do we allow all three htlcs to be nil here? If not, I think we could do switch with cases `htlc != nil` and a default case to error out.

Reply:
The three types, `s.htlcP2WSH`, `s.htlcNP2WSH` and `s.htlcP2TR`.